gp-includes/formats/format-php.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * GlotPress Format PHP class
+ *
+ * @since 4.0.0
+ *
+ * @package GlotPress
+ */
+
+/**
+ * Format class used to support PHP file format.
+ *
+ * @since 4.0.0
+ */
+class GP_Format_PHP extends GP_Format {
+	/**
+	 * Name of file format, used in file format dropdowns.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @var string
+	 */
+	public $name = 'PHP (.php)';
+
+	/**
+	 * File extension of the file format, used to autodetect formats and when creating the output file names.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @var string
+	 */
+	public $extension = 'php';
+
+	/**
+	 * Generates a string the contains the $entries to export in the PHP file format.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param GP_Project         $project         The project the strings are being exported for, not used
+	 *                                            in this format but part of the scaffold of the parent object.
+	 * @param GP_Locale          $locale          The locale object the strings are being exported for. not used
+	 *                                            in this format but part of the scaffold of the parent object.
+	 * @param GP_Translation_Set $translation_set The locale object the strings are being
+	 *                                            exported for. not used in this format but part
+	 *                                            of the scaffold of the parent object.
+	 * @param GP_Translation     $entries         The entries to export.
+	 * @return string The exported PHP string.
+	 */
+	public function print_exported_file( $project, $locale, $translation_set, $entries ) {
+		$language_code = $this->get_language_code( $locale );
+		if ( false === $language_code ) {
+			$language_code = $locale->slug;
+		}
+
+		$current        = $project;
+		$project_tree   = array();
+		$project_tree[] = $current->name;
+
+		while ( $current->parent_project_id > 0 ) {
+			$current        = GP::$project->get( $current->parent_project_id );
+			$project_tree[] = $current->name;
+		}
+
+		$project_tree = array_reverse( $project_tree );
+
+		$project_id_version = implode( ' - ', $project_tree );
+
+		/** This filter is documented in gp-includes/formats/format-pomo.php */
+		$project_id_version = apply_filters( 'gp_pomo_export_project_id_version', $project_id_version, $project_tree );
+
+		$result = array(
+			'x-generator'               => 'GlotPress/' . GP_VERSION,
+			'translation-revision-date' => GP::$translation->last_modified( $translation_set ) . '+0000',
+			'plural-forms'              => "nplurals=$locale->nplurals; plural=$locale->plural_expression;",
+			'project-id-version'        => $project_id_version,
+			'language'                  => $language_code,
+			'messages'                  => array(),
+		);
+
+		/* @var Translation_Entry $entry */
+		foreach ( $entries as $entry ) {
+			$key = $entry->context ? $entry->context . chr( 4 ) . $entry->singular : $entry->singular;
```

Comment:
In the 			
```
$key = $entry->context ? $entry->context . chr( 4 ) . $entry->singular : $entry->singular;
``` 
you are adding the context for the original string. 

Here we can take 2 approaches:

1. Avoid all extra information, using this new format as only an export format, removing the `context`.
2. Add all the elements in the `Translation_Entry`  class:

```
*     @type string $context             A string differentiating two equal strings
*                                       used in different contexts.
*     @type string $translator_comments Comments left by translators.
*     @type string $extracted_comments  Comments left by developers.
*     @type array  $references          Places in the code this string is used, in
*                                       relative_to_root_path/file.php:linenum form.
*     @type array  $flags               Flags like php-format.
```
I think this second approach is the good one, because it will have all the information to use this new format as import format in the future.

Reply:
Hmm not sure about this one.

I see the PHP format more as a lightweight export-only format that doesn't necessarily have to retain all of the metadata.

(1) is not an option. Context is a requirement for being able to translate strings with context, as it makes things unique. You can't omit this. Otherwise how would you store translations for `_x( 'Run', 'noun' )` and `_x( 'Run', 'verb' )`? So this is required.

References and comments on the other hand are optional metadata for translators. Including those in the export by default would significantly increase file size. It would also mean not being able to use `implode()` for the messages, as then it would need to be an array (e.g. `$result['messages'][ $key ] = [ 'translation' => ..., 'comments' => ..., 'references' => ... ]`, which would cause memory overhead as well due to all the array allocations.

In short, this would negate many of the performance benefits of the current, extremely lightweight format.

Reply:
Ok, this explanation makes sense to me, so we will use this new .php format as export-only format.